src/Compilers/CSharp/Test/Emit3/Symbols/UserDefinedCompoundAssignmentOperatorsTests.cs

```diff
@@ -2798,5 +2794,2432 @@ interface I4 : I2
+
+            verifier.VerifyIL("Program.Test1",
+@"
+{
+  // Code size       22 (0x16)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  call       ""C1[] Program.GetA<C1>(C1[])""
+  IL_0006:  call       ""int Program.Get0()""
+  IL_000b:  ldelema    ""C1""
+  IL_0010:  call       ""void C1." + methodName + @"()""
+  IL_0015:  ret
+}
+");
+
+            verifier.VerifyIL("Program.Test3<T>(T[])",
+@"
+{
+  // Code size       30 (0x1e)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  call       ""T[] Program.GetA<T>(T[])""
+  IL_0006:  call       ""int Program.Get0()""
+  IL_000b:  readonly.
+  IL_000d:  ldelema    ""T""
+  IL_0012:  constrained. ""T""
+  IL_0018:  callvirt   ""void I1." + methodName + @"()""
+  IL_001d:  ret
+}
+");
+
+            methodName = (op == "++" ? WellKnownMemberNames.CheckedIncrementOperatorName : WellKnownMemberNames.CheckedDecrementOperatorName);
+
+            verifier.VerifyIL("Program.Test2",
+@"
+{
+  // Code size       22 (0x16)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  call       ""C1[] Program.GetA<C1>(C1[])""
+  IL_0006:  call       ""int Program.Get0()""
+  IL_000b:  ldelema    ""C1""
+  IL_0010:  call       ""void C1." + methodName + @"()""
+  IL_0015:  ret
+}
+");
+
+            verifier.VerifyIL("Program.Test4<T>(T[])",
+@"
+{
+  // Code size       30 (0x1e)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  call       ""T[] Program.GetA<T>(T[])""
+  IL_0006:  call       ""int Program.Get0()""
+  IL_000b:  readonly.
+  IL_000d:  ldelema    ""T""
+  IL_0012:  constrained. ""T""
+  IL_0018:  callvirt   ""void I1." + methodName + @"()""
+  IL_001d:  ret
+}
+");
+
+            comp2 = CreateCompilation(source2, references: [fromMetadata ? comp1.EmitToImageReference() : comp1.ToMetadataReference()], options: TestOptions.DebugExe);
+            verifier = CompileAndVerify(comp2, expectedOutput: @"
+[GetA][Get0][operator]1
+[GetA][Get0][operator checked]2
+[GetA][Get0][operator]3
+[GetA][Get0][operator checked]4
+[GetA][Get0][operator]5
+[GetA][Get0][operator checked]6
+").VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void Increment_076_Consumption_Postfix_For()
+        {
+            var source = @"
+public class C1
+{
+    public int _F;
+    public void operator ++()
+    {
+        _F++;
+    } 
+}
+
+public class Program
+{
+    static void Main()
+    {
+        C1 x = new C1();
+        for (x++; x._F < 4; x++)
+        {
+            System.Console.Write(x._F);
+        }
+    } 
+}
+";
+
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseExe);
+            CompileAndVerify(comp, expectedOutput: @"123").VerifyDiagnostics();
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void Increment_077_Consumption_Postfix_Used([CombinatorialValues("++", "--")] string op, bool fromMetadata)
+        {
+            var source1 = @"
+public class C1
```

Comment:
>class [](http://example.com/codeflow?start=7&length=5)

Are we testing with `struct`, or with type parameter with/without `class` or `struct` constraints?

Reply:
> Are we testing with struct, or with type parameter with/without class or struct constraints?

I do not find these scenarios interesting. The test is not about code that we generate, it is about an error that doesn't depend on the mentioned aspects.